warehouse/tuf/repository.py

```diff
@@ -0,0 +1,497 @@
+        """
+        Loads a specific metadata by role name.
+
+        Args:
+            rolename: Role name
+
+        Returns:
+             Role Metadata
+             ``tuf.api.metadata.Metadata``
+        """
+        return Metadata.from_file(rolename, None, self.storage_backend)
+
+    def delegate_targets_roles(
+        self,
+        payload: Dict[str, List[RolesPayload]],
+    ) -> Metadata[Snapshot]:
+        """
+        Delegates a list of targets roles.
+        It will create the new role based in the parameters, sign and store,
+        updating the Delegator role, Snapshot and Timestamp roles.
+
+        Args:
+            payload: Dictionary of delegators with the delegated role parameters
+                     as RolePayload
+            timestamp_expires: New Timestamp expiration datetime
+            snapshot_expires: New Snapshot expiration datetime
+        """
+
+        snapshot_metadata = self.load_role(Snapshot.type)
+        for delegator, delegate_role_parameters in payload.items():
+            delegator_metadata = self.load_role(delegator)
+            snapshot_metadata = self._create_delegated_targets_roles(
+                delegator_metadata, snapshot_metadata, delegate_role_parameters
+            )
+            delegator_metadata = self.bump_role_version(
+                rolename=delegator,
+                role_metadata=delegator_metadata,
+                role_expires=delegator_metadata.signed.expires,
+                key_rolename=None,
+                store=True,
+            )
+            snapshot_metadata = self.snapshot_update_meta(
+                delegator, delegator_metadata.signed.version, snapshot_metadata
+            )
+
+        return snapshot_metadata
+
+    def bump_role_version(
+        self,
+        rolename: str,
+        role_metadata: Metadata,
+        role_expires: datetime,
+        key_rolename: Optional[str] = None,
+        store: Optional[bool] = False,
+    ) -> Metadata:
+        """
+        Bump an specific Role Metadata (ex: delegated target role) version,
+        optionaly stores it in the repository storage.
+
+        Args:
+            rolename: Specific Role name to have the version bumped
+            role_metadata: Specific Role Metadata
+            role_expires: expiration datetime
+            key_rolename: Different signing Key role name
+            store: Optional, Default False, stores in the repository storage
+        """
+        if key_rolename:
+            key_rolename = key_rolename
+        else:
+            key_rolename = rolename
+        role_metadata.signed.expires = role_expires
+        role_metadata.signed.version += 1
+        key_rolename_keys = self.key_backend.get(key_rolename, "private")
+        for key in key_rolename_keys:
+            role_metadata.sign(SSlibSigner(key), append=True)
+
+        if store:
+            self._store(rolename, role_metadata)
+
+        return role_metadata
+
+    def timestamp_bump_version(
```

Comment:
Looks like this one could just call `bump_role_version`?

Reply:
It makes me confused because the specific `timestamp_metadata.signed.snapshot_meta = MetaFile(version=snapshot_version)`
IMO it looks very 'explicit' not sure

Reply:
Yeah, I guess that's okay.